main.js

```diff
@@ -1,112 +0,0 @@
-/**
- * Created by Jenya on 18.04.15.
- */
-
-/// At first I'm creating general BeingClass. It will be the top of all being classes.
-function BeingClass(clas, name, legsAmount) {
-    this.clas = clas;
-    this.name = name;
-    this.legs = legsAmount;
-    this.movementType = "";
-}
-
-// Than I'm adding move function to all beings.
-BeingClass.prototype.makeMove = function () {
-    this.movementType.move(this.name);
-    return this;
-};
-
-BeingClass.prototype.setLegsAmount = function (legsAmount) {
-    this.legs = legsAmount;
-    return this;
-};
-
-//Making a method to print object's properties
-BeingClass.prototype.logName = function () {
-    console.log("This.class: " + this.clas + "; this.name: " + this.name + "; this.legs= " + this.legs);
-    return this;
-};
-
-//Now I'm setting movement methods, strategy and type switch - BEGIN
-BeingClass.prototype.setMovementType = function (newMovementType) {
-    this.movementType = newMovementType;
-    return this;
-};
-
-var move = {};
-
-var Movement = function (func) {
-    this.move = func;
```

Comment:
The `Movement` constructor was the source of this "complication".
as it assigned `func` to `this`**`.move`** instead of the callee itself.
Changed it to `return func` and everything works properly without the `.move` that disturbed you.